### ISSUE TITLE
Upgrade peer dependencies react & react-dom to version 16.0.0

### ISSUE THREAD

**Issue by ajhool** [neptunian/react-photo-gallery#71]
Currently, the react-dom peer dependencies are behind the devDependencies -- and all of them are a few versions behind the current stable react-dom release.

from package.json (Note the 15.6.1 and 15.5.0 mismatch):
```
 "devDependencies":{
   ... 
->  "react": "^15.6.1",
    "react-component-gulp-tasks": "^0.7.7",
->  "react-dom": "^15.6.1",
    "react-images": "~0.5.2",
    "react-measure": "^2.0.2"
  },
  "peerDependencies": {
->  "react": "~15.5.0",
->  "react-dom": "~15.5.0"
  },
```

Additionally, since the 15.6.1 version, there have been two react-dom & react releases:
15.6.2
16.0.0 

I'm currently tracking down a few bugs and am not sure if this is their cause -- but will update when I figure them out!

**EDIT: The other bugs were related to separate package management issues on my machine -- I was working with an old version without knowing it. A lot of other packages are behind on the peer dependencies, so this issue is not pressing but is probably worth getting to at some point**

**Comment by neptunian** [comment]
thanks @ajhool i see react 16 was released.. i'll fix this when i do some tests and upgrade, thanks!

**Comment by neptunian** [comment]
This should be fixed in 6.0.7
